src/enum.lisp

```diff
@@ -78,6 +78,16 @@
           (ensure-list pair)
         (check-type keyword enum-key)
         ;;(check-type value integer)
+        ;; resolve field reference
+        (when (keywordp value)
+          (setf value (gethash value keyword-values)))
+        (when (listp value)
+          (setf value (eval
+                       (list 'let (loop for k
+                                        being the hash-key
+                                        using (hash-value v) of keyword-values
+                                        collect (list (intern (string k)) v))
+                             value))))
```

Comment:
What's going on here in this `(when (listp value) ...)` branch?

Reply:
This evaluates the value of enum field in cases like ```(:UDAT_FULL_RELATIVE (cl:logior UDAT_FULL UDAT_RELATIVE))```.
The code creates (let ...) binding current values of :UDAT_FULL and :UDAT_RELATIVE to variables UDAT_FULL and  UDAT_RELATIVE correspondingly and to evaluate (cl:logior UDAT_FULL UDAT_RELATIVE).

Reply:
I see. Did you consider computing these derived values in SWIG when generating the bindings? These seems like implementation details leaking into the binding. Nevertheless, I can see how this might be convenient even for writing bindings by hand.

This approach is problematic though, as field names can collide with the names of Lisp constants in which case evaluating that `let` form will yield an error.